packetbeat/protos/http/http_parser.go

```diff
@@ -160,20 +162,23 @@ func (*parser) parseHTTPLine(s *stream, m *message) (cont, ok, complete bool) {
 		}
 	} else {
 		// REQUEST
-		slices := bytes.Fields(fline)
-		if len(slices) != 3 {
+		afterMethodIdx := bytes.IndexFunc(fline, unicode.IsSpace)
+		afterRequestURIIdx := bytes.LastIndexFunc(fline, unicode.IsSpace)
+
+		// Make sure we have the VERB + URI + HTTP_VERSION
+		if afterMethodIdx == -1 || afterRequestURIIdx == -1 || afterMethodIdx == afterRequestURIIdx {
 			if isDebug {
 				debugf("Couldn't understand HTTP request: %s", fline)
 			}
 			return false, false, false
 		}
 
-		m.method = common.NetString(slices[0])
-		m.requestURI = common.NetString(slices[1])
+		m.method = common.NetString(fline[:afterMethodIdx])
+		m.requestURI = common.NetString(fline[afterMethodIdx+1 : afterRequestURIIdx])
```

Comment:
if there is more than one space between URI and HTTP_VERSION it will be stored into requestURI. I think you should check for that (bytes.Fields did skip consecutive separators)

Reply:
I gave some thoughts about it, I don't think we should check for that, if we do it it would mean that we could _potentially normalize_ the query string parameters. 

If we look at the following case:
```
GET /e/b?movie=a brand new world HTTP/1.1 => '/e/b?movie=a brand new world'
GET /e/b?movie=a brand new world  HTTP/1.1 =>  
/elastic/beats?movie=a brand new world '
```

When we extract the parameters that should gives us 2 distinct values for the `movie` field: `a brand new world` and `a brand new world ` (not the space at the end of world)

WDYT?

Reply:
I ran a quick check and the server did skip those extra spaces.

Now I was testing again and see that it's more common for servers to return a 400 Bad Request.

So I guess you can just ignore the comment. It's not common at all and spaces are escaped to %20 or + anyway